money-java-servlet/src/main/scala/com/comcast/money/java/servlet/TraceFilter.scala

```diff
@@ -33,35 +33,69 @@ import scala.util.{ Failure, Success }
 class TraceFilter extends Filter {
 
   private val MoneyTraceHeader = "X-MoneyTrace"
+  private val B3TraceId = "X-B3-TraceId"
+  private val B3SpanId = "X-B3-SpanId"
+  private val B3ParentSpanId = "X-B3-ParentSpanId"
   private val logger = LoggerFactory.getLogger(classOf[TraceFilter])
   private val factory = Money.Environment.factory
 
   override def init(filterConfig: FilterConfig): Unit = {}
 
   override def destroy(): Unit = {}
 
+  private val spanName = "servlet"
+
   override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
 
     SpanLocal.clear()
     val httpRequest = new HttpServletRequestWrapper(request.asInstanceOf[HttpServletRequest])
-    val incomingTraceId = Option(httpRequest.getHeader(MoneyTraceHeader)) map { incTrcaceId =>
+
+    val maybeMoney = Option(httpRequest.getHeader(MoneyTraceHeader))
```

Comment:
I think that the logic is a little hard to follow here.

The process should be:

1. Attempt to get parse the money header, fall back to the b3 header
2. Set the SpanLocal if either of the headers is present
3. Always just force copy over the request header to the response header.

```scala
def parseMoneyTraceHeader(httpRequest: HttpRequest): Option[SpanId] = ???

def parseB3TraceHeader(httpRequest: HttpRequest): Option[SpanId] = ???

```

With those two helpers, we can attempt to pull out the passed in span id

```scala
val spanId: Option[SpanId] = parseMoneyTraceHeader(httpRequest).orElse(parseB3TraceHeader)
spanId.foreach(s => SpanLocal.push(factory.newSpan(s, spanName)))
```

Then, on the way out, all we need to do is duplicate the request header...

```scala
Option(httpRequest.getHeader(Money)).foreach(response.setHeader(...))
Option(httpRequest.getHeader(B3TraceId)).foreach(response.setHeader))
Option(httpRequest.getHeader(B3ParentId)).foreach...
Option(httpRequest.getHeader(B3SpanId)).foreach...

// given the repetition above, would probably make sense to have a simple helper function like 

def setResponseHeader(headerName): Unit = Option(httpRequest.getHeader).foreach(httpResponse.setHeader...)

```

Reply:
Looks like we may keep the `Try`, in which case we can still follow this.  The `parseMoneyTraceHeader` would attempt to parse (if not there it would be none).  If there, get the try, log any errors, and just do a `toOption`.

We do want to log I think, but not fail the request.